kin-devplatform-sdk/src/main/java/kin/devplatform/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -195,6 +204,17 @@ public Balance getBalance() {
 
 	@Override
 	public void getBalance(@Nullable final KinCallback<Balance> callback) {
+		if (account == null) {
+			if (callback != null) {
+				mainThread.execute(new Runnable() {
+					@Override
+					public void run() {
+						callback.onFailure(ErrorUtil.getClientException(ClientException.ACCOUNT_NOT_LOGGED_IN, null));
```

Comment:
How is it possible for  them to call this method before they are logged in?

Reply:
it's not, removed, I revert most of `BlockchainSourceImpl` changes as they're not needed after uniting to single start method.